core/src/main/java/org/apache/iceberg/ManifestOutputFileFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.iceberg.io.OutputFile;
+
+class ManifestOutputFileFactory {
```

Comment:
Question: Do we really need this class given that `SnapshotProducer` previously kept the manifest count? Can we consider using `Supplier<OutputFile> outputFileSupplier` as an argument to the rolling writer?

Reply:
I want to avoid using the lambda function. It should be OK to use a `Supplier`.